src/lib/ThemeContext.tsx

```diff
@@ -92,33 +93,32 @@ export const ThemeContext = createContext<Theme>(defaultTheme)
 
 export const useTheme = () => useContext(ThemeContext)
 
+const getTheme = (theme: ColorSchemeName) =>
+  theme === 'dark' ? darkTheme : defaultTheme
+
 export const ThemeProvider: React.FC<ThemeProviderProps> = ({
   theme,
   children,
 }) => {
   const colorSchemeFromRN = useColorScheme()
-  const [nativeColorScheme, setNativeColorScheme] =
+  const [statefulColorScheme, setStatefulColorScheme] =
     React.useState<ColorSchemeName>(colorSchemeFromRN)
+  const colorScheme = isWeb ? colorSchemeFromRN : statefulColorScheme
+  const themeValue = getTheme(theme === 'system' ? colorScheme : theme)
 
   React.useEffect(() => {
+    if (!isWeb) return
```

Comment:
I find the use of negation a bit tricky to follow here. Why do we exit early when we're _not_ on the web? Isn't this block of code relevant *when* we're not on the web (i.e. specifically for iOS)? 

If I'm not wrong, we should flip this to `if (isWeb) return`.

As a nitpick these conditions are tricky to think about so I'd prefer less use of early returns here. (Curious if you disagree!) Assuming my earlier suggestion was correct:

```js
  React.useEffect(() => {
    // With RN iOS, we can only "trust" the color scheme reported while the app is active.
    // This bug will probably be fixed by https://github.com/facebook/react-native/pull/39439
    if (!isWeb) {
      const subscription = AppState.addEventListener('change', state => {
        if (state === 'active') {
          setStatefulColorScheme(colorSchemeFromRN)
        }
      })
      return () => subscription.remove()
    }
  }, [colorSchemeFromRN])
```

Reply:
This is also nitpicking, but I'll throw in a few more naming habits I picked up from FB as a proposal.

It might be nice to import the original as `useColorScheme_BUGGY` and then extract this part into a separate `useColorScheme_FIXED` Hook which encapsulates this fix. Then it'll be easy to get rid of the forked behavior without making mistakes when https://github.com/facebook/react-native/pull/39439 lands into a release we're using.

```js
import {
  // ...
  useColorScheme as useColorScheme_BUGGY,
  // ...
} from 'react-native'

// ...

const useColorScheme_FIXED = () => {
  const colorSchemeFromRN = useColorScheme_BUGGY()
  const [savedColorScheme, setSavedColorScheme] = useState<ColorSchemeName>(colorSchemeFromRN)
  const colorScheme = isWeb ? currentScheme : savedColorScheme
  // ...
  return colorScheme
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({
  // ...
}) => {
  const colorScheme = useColorScheme_FIXED()
  // ...
```

Then when the bug is fixed, we'll remove alias from the import, delete `useColorScheme_FIXED`, and call the imported `useColorScheme` directly.

Reply:
>If I'm not wrong

You're not wrong 🤦 good catch

Reply:
I like the hook idea too, let's adopt that throughout 👍